infrastructure/libtor/src/tor.rs

```diff
@@ -95,15 +90,36 @@ impl Tor {
         instance.passphrase = TorPassword(Some(passphrase));
 
         // data dir
-        let temp = tempdir()?;
-        instance.data_dir = temp.path().to_path_buf();
-        instance.temp_dir = Some(temp);
+        instance.data_dir = data_dir.join("data");
+        match fs::exists(instance.data_dir.clone()) {
+            Ok(exists) => {
+                if !exists {
+                    if let Err(e) = fs::create_dir_all(&instance.data_dir) {
+                        return Err(ExitError::new(
+                            ExitCode::InputError,
+                            format!(
+                                "Could not create libtor data directory: {} ({})",
+                                instance.data_dir.display(),
+                                e
+                            ),
+                        ));
+                    }
+                }
+            },
+            Err(e) => {
+                return Err(ExitError::new(
+                    ExitCode::InputError,
+                    format!(
+                        "Could not verify libtor data directory: {} ({})",
+                        instance.data_dir.display(),
+                        e
+                    ),
+                ));
+            },
+        }
```

Comment:
nit: manual implementation of map_err
```suggestion
        let exists = fs::exists(instance.data_dir.clone()).map_err(|e| ExitError::new(
                    ExitCode::InputError,
                    format!(
                        "Could not verify libtor data directory: {} ({})",
                        instance.data_dir.display(),
                        e
                    ),
                ))?;

                if !exists {
                    fs::create_dir_all(&instance.data_dir).map_err(|e| ExitError::new(
                            ExitCode::InputError,
                            format!(
                                "Could not create libtor data directory: {} ({})",
                                instance.data_dir.display(),
                                e
                            ),
                        ))?;
                }
           
```